pkg/parsers/parsers.go

```diff
@@ -74,26 +75,33 @@ func ParseTCPAddr(tryAddr string, defaultAddr string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	hostParts := strings.Split(u.Host, ":")
-	if len(hostParts) != 2 {
+
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
 		return "", fmt.Errorf("Invalid bind address format: %s", tryAddr)
 	}
-	defaults := strings.Split(defaultAddr, ":")
-	if len(defaults) != 3 {
-		return "", fmt.Errorf("Invalid defaults address format: %s", defaultAddr)
+
+	defaultAddr = strings.TrimPrefix(defaultAddr, "tcp://")
+	defaultHost, defaultPort, err := net.SplitHostPort(defaultAddr)
+	if err != nil {
+		return "", err
 	}
 
-	host := hostParts[0]
 	if host == "" {
-		host = strings.TrimPrefix(defaults[1], "//")
+		host = defaultHost
 	}
-	if hostParts[1] == "" {
-		hostParts[1] = defaults[2]
+	if port == "" {
+		port = defaultPort
 	}
-	p, err := strconv.Atoi(hostParts[1])
+	p, err := strconv.Atoi(port)
 	if err != nil && p == 0 {
 		return "", fmt.Errorf("Invalid bind address format: %s", tryAddr)
 	}
+
+	if net.ParseIP(host).To4() == nil && strings.Contains(host, ":") {
```

Comment:
I'm not sure why I would want to join here.
I'm verifying that the IP is indeed an ipv6 address, I don't want the port.